### PR TITLE
Adopt GitHub Actions for CI

### PR DESCRIPTION
**Pull request by davidfurey**
## What does this change?

This change adopt GitHub Actions for CI, replacing TeamCity which DevX are switching off in Q2 2023.

This change isn't exactly a no-op switch, as the [`sbt-riffraff-artifact` plugin](https://github.com/guardian/sbt-riffraff-artifact) has been removed in favour of [`guardian/actions-riff-raff`](https://github.com/guardian/actions-riff-raff). 

## How to test
- [x] Deploy to CODE

## Notes
- I've paused the [TeamCity project](https://teamcity.gutools.co.uk/buildConfiguration/EditorialTools_EditorialPreferences?mode=branches).

`.deb` filesize is different between build 700 and build 701, but unzipping the deb yields no differences so I presume this is a quirk of the zip compression

### CODE REVIEW

**Comment by davidfurey** [comment]
I've done a cursory test in CODE, adding and removing an article from a story package

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+              with:
+                  node-version-file: '.nvmrc'
+                  cache: npm
+                  cache-dependency-path: 'package-lock.json'
+
+            - name: Setup Java
+              uses: actions/setup-java@v3
+              with:
+                  java-version: '8'
+                  distribution: 'corretto'
```
Comment (samanthagottlieb) [inline .github/workflows/ci.yml:38]: Adding `cache: 'sbt'` here is apparently helpful for performance, I learnt [here](https://github.com/guardian/grid/pull/4146#discussion_r1346060792).

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            - main
+jobs:
+    CI:
+        runs-on: ubuntu-latest
+        permissions:
+            # required by aws-actions/configure-aws-credentials
+            id-token: write
+            contents: read
+        steps:
+            - uses: actions/checkout@v3
```
Comment (samanthagottlieb) [inline .github/workflows/ci.yml:21]: There's a [v4](https://github.com/actions/checkout/releases) of this action.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+            - uses: aws-actions/configure-aws-credentials@v2
```
Comment (samanthagottlieb) [inline .github/workflows/ci.yml:23]: Also a [v4](https://github.com/aws-actions/configure-aws-credentials/releases) of this.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                  jspm registry export github
+                  jspm install
+
+            - name: Grunt
+              run: |
+                  grunt --stack validate test
+                  grunt --stack bundle
+
+            - name: SBT
+              run: sbt "clean;compile;test;assets;Debian / packageBin"
```
Comment (samanthagottlieb) [inline .github/workflows/ci.yml:65]: Just a note on batch mode, apparently it makes builds run [more slowly](https://www.scala-sbt.org/1.x/docs/offline/Running.html#Batch+mode). Having said that, [this discussion](https://github.com/guardian/grid/pull/4146#discussion_r1344340943) suggests that these sbt commands will run in batch mode, and sequentially (i.e. no need for semicolons either), anyway.

I believe you can also remove the space and have `Debian/packageBin`, to remove the need for quotes.

**Review by samanthagottlieb — APPROVED** [review]
Looks good!

**Comment by davidfurey** [comment]
Thanks @samanthagottlieb, I'll merge+deploy tomorrow

**Pull request merged**